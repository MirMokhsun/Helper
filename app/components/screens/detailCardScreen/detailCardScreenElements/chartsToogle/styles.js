import { StyleSheet } from 'react-native';

export const styles = StyleSheet.create({
    wrapperToogle: {
        width: '100%',
        paddingHorizontal: 16,
        paddingVertical: 8
    },
    containerToogle: {
        width: 130,
        height: 32,
        backgroundColor: '#EBECEE',
        borderRadius: 16,
        padding: 4
    },
    wrapperImageInToogle: {
        position: 'absolute',
        top: 4,
        left: 4,
        width: '100%',
        height: '100%',
        flexDirection: 'row',
        zIndex: 2,
    },
    sliderInToogle: {
        width: '50%',
        height: 24,
        borderRadius: 12,
        backgroundColor: '#FFFFFF'
    },
    imageInToogle: {
        width: '50%',
        justifyContent: 'center',
        alignItems: 'center',
    },

});