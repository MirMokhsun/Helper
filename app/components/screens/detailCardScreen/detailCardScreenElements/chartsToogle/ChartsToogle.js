import React, { PureComponent } from 'react';
import { View, Image, Animated, TouchableWithoutFeedback, Easing, I18nManager } from 'react-native';
import { styles } from './styles';

const lineChartBlue = <Image source={require('./iconPNG/blueLine.png')} style={{ width: 26, height: 18, marginRight: 4 }} />;
const candleChartGray = <Image source={require('./iconPNG/grayCandle.png')} style={{ width: 14, height: 18, marginLeft: 4 }} />;
const lineChartGray = <Image source={require('./iconPNG/grayLine.png')} style={{ width: 26, height: 18, marginRight: 4 }} />;
const candleChartBlue = <Image source={require('./iconPNG/blueCandle.png')} style={{ width: 14, height: 18, marginLeft: 4 }} />;

export class ChartsToogle extends PureComponent {
    constructor(props) {
        super(props);
        this.maxLeftDistance = I18nManager.isRTL ? -61 : 61;
        this.state = {
            leftDistance: new Animated.Value(this.maxLeftDistance),
            isCandleChartsChosen: false,
            lineChart: lineChartGray,
            candleChart: candleChartBlue,
        }
    }

    animation = (_toDistance, _fromDistane) => {
        const { leftDistance } = this.state;
        Animated.timing(
            leftDistance,
            {
                fromValue: _fromDistane,
                toValue: _toDistance,
                duration: 100,
                easing: Easing.linear,
                useNativeDriver: true
            }
        ).start();
    }

    changeChart = async () => {
        const { changeCharts } = this.props;
        const { isCandleChartsChosen } = this.state;
        let lineChart; let candleChart = null;

        if (isCandleChartsChosen) {
            lineChart = lineChartGray;
            candleChart = candleChartBlue;
            this.animation(this.maxLeftDistance, 0);
        } else {
            lineChart = lineChartBlue;
            candleChart = candleChartGray;
            this.animation(0, this.maxLeftDistance);
        }
        this.setState({
            isCandleChartsChosen: !isCandleChartsChosen,
            lineChart,
            candleChart,
        });
        await changeCharts();
    };

    render() {
        const { lineChart, candleChart, leftDistance } = this.state;
        return (
            <TouchableWithoutFeedback onPress={this.changeChart} testID="ChartsToogleID" accessibilityLabel="ChartsToogleID">
                <View style={styles.containerToogle} >
                    <Animated.View style={[styles.sliderInToogle, { transform: [{ translateX: leftDistance }] }]} ></Animated.View>
                    <View style={styles.wrapperImageInToogle}>
                        <View style={styles.imageInToogle}>{lineChart}</View>
                        <View style={styles.imageInToogle}>{candleChart}</View>
                    </View>
                </View>
            </TouchableWithoutFeedback>
        );
    }
}

export default ChartsToogle;