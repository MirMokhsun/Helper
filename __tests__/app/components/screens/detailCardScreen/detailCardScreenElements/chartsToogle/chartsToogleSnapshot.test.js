import 'react-native';
import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import { ChartsToogle } from '../../../../../../../app/components/screens/detailCardScreen/detailCardScreenElements/chartsToogle/ChartsToogle';

it('create snapshot AskSpredBidLine', () => {
    const renderer = new ShallowRenderer()
    let toolbarSnapshot = renderer.render(<ChartsToogle />);
    expect(toolbarSnapshot).toMatchSnapshot();
});