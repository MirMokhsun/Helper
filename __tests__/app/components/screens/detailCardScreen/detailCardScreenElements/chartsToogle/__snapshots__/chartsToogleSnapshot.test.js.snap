// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`create snapshot AskSpredBidLine 1`] = `
<TouchableWithoutFeedback
  accessibilityLabel="ChartsToogleID"
  onPress={[Function]}
  testID="ChartsToogleID"
>
  <View
    style={
      Object {
        "backgroundColor": "#EBECEE",
        "borderRadius": 16,
        "height": 32,
        "padding": 4,
        "width": 130,
      }
    }
  >
    <AnimatedComponent
      style={
        Array [
          Object {
            "backgroundColor": "#FFFFFF",
            "borderRadius": 12,
            "height": 24,
            "width": "50%",
          },
          Object {
            "transform": Array [
              Object {
                "translateX": 61,
              },
            ],
          },
        ]
      }
    />
    <View
      style={
        Object {
          "flexDirection": "row",
          "height": "100%",
          "left": 4,
          "position": "absolute",
          "top": 4,
          "width": "100%",
          "zIndex": 2,
        }
      }
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "50%",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../app/components/screens/detailCardScreen/detailCardScreenElements/chartsToogle/iconPNG/grayLine.png",
            }
          }
          style={
            Object {
              "height": 18,
              "marginRight": 4,
              "width": 26,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "width": "50%",
          }
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../app/components/screens/detailCardScreen/detailCardScreenElements/chartsToogle/iconPNG/blueCandle.png",
            }
          }
          style={
            Object {
              "height": 18,
              "marginLeft": 4,
              "width": 14,
            }
          }
        />
      </View>
    </View>
  </View>
</TouchableWithoutFeedback>
`;
